### PR TITLE
AMD Context fix

### PR DESCRIPTION
**Pull request by scotthasbrouck**
@dmitrig01 Tested this in a sample reqirejs page, and it works great. Looks like this is the only change that needs to happen.

### CODE REVIEW

**Comment by dmitrig01** [comment]
so if we include the js file on a page, we can later `require(['branch'], function(branch) { ... })` or something like thtat?

**Comment by dmitrig01** [comment]
maybe we shoudl docuent it :P

**Comment by dmitrig01** [comment]
@scotthasbrouck did you look into module loading techniques and come up with something that'll work for everyone, as discussed?

File: src/4_initialization.js [hunk 1]
```diff
@@ -23,12 +23,13 @@ if (window['branch'] && window['branch']['_q']) {
 
 // AMD
 if (typeof define === 'function' && define.amd) {
-	define('branch', function() { return branch_instance; });
+	define(['branch'], function() { return branch_instance; });
```
Comment (dmitrig01) [inline src/4_initialization.js]: very minor code style thing - i think the style is to put a space around the quoted string - not sure why the linter didn't catch it

**Comment by scotthasbrouck** [comment]
Found the .jscs paramater that requires spaces inside arrays, so the linter should catch that going forward. I still need to ask the guy to clarify what problem he's having, because it all seems to work for me. I'll do that now

**Comment by scotthasbrouck** [comment]
actually, do you want spaces around strings _just_ inside of array brackets, or all brackets? There's a parameter for that also

**Comment by dmitrig01** [comment]
array and curly

**Comment by dmitrig01** [comment]
but really, most important is to make a rule and keep consistent :-)

**Comment by dmitrig01** [comment]
i hate to nitpick on this, but what i meant (:-/) was: spaces inside single-line array literals and object literals, and no spaces inside the rest

what we use on backend

```
{
	"fileExtensions": [".js", ".jsx"],

	"requireCurlyBraces": [ "if", "else", "for", "while", "do", "try", "catch" ],
	"requireSpaceAfterKeywords": [ "if", "else", "for", "while", "do", "switch", "return", "try", "catch" ],
	"requireSpaceBeforeBlockStatements": true,
	"requireParenthesesAroundIIFE": true,
	"requireSpacesInConditionalExpression": { "afterTest": true, "beforeConsequent": true, "afterConsequent": true, "beforeAlternate": true },
	"requireSpacesInFunctionExpression": { "beforeOpeningCurlyBrace": true },
	"disallowSpacesInFunctionExpression": { "beforeOpeningRoundBrace": true },
	"requireSpacesInAnonymousFunctionExpression": { "beforeOpeningCurlyBrace": true },
	"disallowSpacesInAnonymousFunctionExpression": { "beforeOpeningRoundBrace": true },
	"requireSpacesInNamedFunctionExpression": { "beforeOpeningCurlyBrace": true },
	"disallowSpacesInNamedFunctionExpression": { "beforeOpeningRoundBrace": true },
	"requireSpacesInFunctionDeclaration": { "beforeOpeningCurlyBrace": true },
	"disallowSpacesInFunctionDeclaration": { "beforeOpeningRoundBrace": true },
	"requireSpacesInFunction": { "beforeOpeningCurlyBrace": true },
	"disallowSpacesInFunction": { "beforeOpeningRoundBrace": true },
	"disallowSpacesInsideParentheses": true,
	"requireSpacesInsideObjectBrackets": "all",
	"requireSpacesInsideArrayBrackets": "all",
	"disallowSpaceAfterObjectKeys": true,
	"requireCommaBeforeLineBreak": true,
	"disallowSpaceAfterPrefixUnaryOperators": ["++", "--", "+", "-", "~", "!"],
	"disallowSpaceBeforePostfixUnaryOperators": ["++", "--", ";"],
	"disallowSpaceBeforeBinaryOperators": [ "," ],
	"requireSpaceBeforeBinaryOperators": [ "=", "+", "-", "/", "*", "==", "===", "!=", "!==" ],
	"requireSpaceAfterBinaryOperators": [ "=", ",", "+", "-", "/", "*", "==", "===", "!=", "!==" ],
	"disallowMultipleLineStrings": true,
	"validateLineBreaks": "LF",
	"disallowMixedSpacesAndTabs": "smart",
	"disallowTrailingWhitespace": true,
	"disallowTrailingComma": true,
	"requireKeywordsOnNewLine": [ "else" ],
	"requireLineFeedAtFileEnd": true,
	"requireCapitalizedConstructors": true,
	"safeContextKeyword": [ "self" ],
	"disallowYodaConditions": true,
	"requireSpaceAfterLineComment": true,
	"disallowNewlineBeforeBlockStatements": true
}
```

**Comment by scotthasbrouck** [comment]
Ah, ok. Any reason I can't just use this? Minus the `. jsx` in `fileExtensions`

**Comment by Kirkkt** [comment]
@scotthasbrouck  I will close this PR in 7 days because it's a bit outdated. Let me know if you think otherwise.

**Comment by dmitrig01** [comment]
@Kirkkt this is still relevant - can you take on testing in AMD/RequireJS and making sure it works

**Comment by jsaleigh** [comment]
This fix breaks webpack

File: src/4_initialization.js [hunk 2]
```diff
@@ -23,12 +23,13 @@ if (window['branch'] && window['branch']['_q']) {
 
 // AMD
 if (typeof define === 'function' && define.amd) {
-	define('branch', function() { return branch_instance; });
+	define([ 'branch' ], function() { return branch_instance; });
```
Comment (jsaleigh) [inline src/4_initialization.js]: This would create an anonymous module that requires 'branch', instead of creating a 'branch' module.

**Comment by aaustin** [comment]
Closing this as well.

**Pull request closed without merging**